hypothesis-python/src/hypothesis/extra/cli.py

```diff
@@ -104,6 +107,68 @@ def obj_name(s: str) -> object:
                 + (f"  Closest matches: {matches!r}" if matches else "")
             )
 
+    def _refactor(func, fname):
+        try:
+            with open(fname) as f:
+                oldcode = f.read()
+        except (OSError, UnicodeError) as err:
+            # Permissions or encoding issue, or file deleted, etc.
+            return f"skipping {fname!r} due to {err}"
+        newcode = func(oldcode)
+        if newcode != oldcode:
+            with open(fname, mode="w") as f:
+                f.write(newcode)
+
+    @main.command()  # type: ignore  # Click adds the .command attribute
+    @click.argument("path", type=str, required=True, nargs=-1)
+    def codemod(path):
+        """`hypothesis codemod` refactors deprecated or inefficent code.
+
+        It adapts `python -m libcst.tool`, removing many features and config options
+        which are rarely relevant for this purpose.  If you need more control, we
+        encourage you to use the libcst CLI directly; if not this one is easier.
+
+        PATH is the file(s) or directories of files to format in place, or
+        "-" to read from stdin and write to stdout.
+        """
+        try:
+            from libcst.codemod import gather_files
+
+            from hypothesis.extra import codemods
+        except ImportError:
+            sys.stderr.write(
+                "You are missing required dependencies for this option.  Run:\n\n"
+                "    python -m pip install --upgrade hypothesis[codemods]\n\n"
+                "and try again."
+            )
+            sys.exit(1)
+
+        # Special case for stdin/stdout usage
+        if any(p == "-" for p in path):
```

Comment:
I feel I’m missing something; why not do:

```suggestion
        if "-" in path:
```

Reply:
Excellent suggestion - I'd uncritically imitated LibCST, but your suggestion is definitely nicer 😄